src/api/queues.js

```diff
@@ -41,16 +48,19 @@ router.get(
       'questionCount'
     ).findAll()
 
-    const filteredCourseIds = await Course.findAll()
-      .then(course =>
-        course.filter(
-          c =>
-            userAuthz.isAdmin ||
-            !c.isUnlisted ||
-            userAuthz.staffedCourseIds.indexOf(c.id) !== -1
-        )
-      )
-      .map(course => course.id)
+    let courses
+    if (userAuthz.isAdmin) {
+      courses = await Course.findAll()
+    } else {
+      const { staffedCourseIds } = userAuthz
+      courses = await Course.findAll({
```

Comment:
Would it be possible to express this filtering directly in the queues query? Something like:

```js
Queue.scope(...).findAll({
  include: [{
    model: Course,
    where: {
      [Sequelize.Op.or]: [{ id: staffedCourseIds }, { isUnlisted: false }],
    }
  }]
}
```

In general, any kind of filtering like this will be much more efficient when done by the query engine.

Reply:
Gotcha, will do!